packages/jest-editor-support/src/Settings.js

```diff
@@ -58,16 +58,21 @@ module.exports = class Settings extends EventEmitter {
     this.debugprocess.stdout.on('data', (data: Buffer) => {
       const string = data.toString();
       // We can give warnings to versions under 17 now
+      // See https://github.com/facebook/jest/issues/2343 for moving this into 
+      // the config object
       if (string.includes('jest version =')) {
-        const version = string.split('jest version =').pop().split(EOL)[0];
-        this.jestVersionMajor = parseInt(version, 1);
+        const version = string.split('jest version =')
+          .pop()
+          .split(EOL)[0]
+          .trim();
+        this.jestVersionMajor = parseInt(version, 10);
       }
       
       // Pull out the data for the config
       if (string.includes('config =')) {
         const jsonString = string.split('config =')
```

Comment:
I think we should keep the breaking changes to a minimum, but current-jest - 1 is reasonable unless you guys are willing to support older versions on your own (I'm happy to merge).

I'm ok having an integration test for the current version on master as a start; I care most about making sure we don't break what's there currently and it should be simple to add one that runs jest-editor-support directly and calls Jest to make sure it doesn't crash.